app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,182 @@
+class OrdersController < ApplicationController
+  before_action :require_login, only: [:index, :pending, :cancelled, :paid, :completed, :show]
+  before_action :get_order, only: [:confirm, :show]
+  before_action :get_order_items, only: [:index, :status]
+  before_action :get_order_item_revenue, only: :show
+
+  def checkout
+    # get @current_order info from carts controller
+    @cart_items = session[:cart]
+    # @cart_items looks like { "1" => 2 }
```

Comment:
thanks for this comment! It really helped me understand why you called k.to_i on Line 17 :heart